charts/dns/templates/pdns-deployment.yaml

```
@@ -6,65 +6,125 @@ metadata:
-          ports:
-          {{- if .Values.pdns.httpService.enabled }}
-          - containerPort: {{ .Values.pdns.httpService.port }}
-            name: http
-            protocol: TCP
-          {{- end }}
-          {{- if .Values.pdns.udpService.enabled }}
-          - containerPort: {{ .Values.pdns.udpService.port }}
-            name: dns-udp
-            protocol: UDP
-          {{- end }}
-          {{- if .Values.pdns.tcpService.enabled }}
-          - containerPort: {{ .Values.pdns.tcpService.port }}
-            name: dns-tcp
-            protocol: TCP
+            {{- toYaml . | nindent 12 }}
           {{- end }}
+          command:
+            - /bin/bash
+            - /entrypoint/script
+            - /usr/local/sbin/pdns_server
+          args:
+            - --launch=gsqlite3
+            - --gsqlite3-database=$(DATABASE_PATH)
+
+            {{- if .Values.service.http.enabled }}
+            - --webserver=yes
+            - --webserver-address=0.0.0.0
+            - --webserver-port={{ .Values.service.http.port }}
+            - --api=yes
+            - --api-key=$(POWERDNS_API_KEY)
+            {{- end }}
+
+            {{- range $arg := .Values.pdns.extraArgs }}
+            - {{ $arg }}
+            {{- end }}
+          {{- if or .Values.service.http.enabled (or .Values.service.udp.enabled .Values.service.tcp.enabled) }}
+          ports:
+            {{- if .Values.service.tcp.enabled }}
+            - name: dns-tcp
+              protocol: TCP
+              containerPort: {{ .Values.service.tcp.port }}
+            {{- end }}
+            {{- if .Values.service.udp.enabled }}
+            - name: dns-udp
+              protocol: UDP
+              containerPort: {{ .Values.service.udp.port }}
+            {{- end }}
+            {{- if .Values.service.http.enabled }}
+            - name: http
+              protocol: TCP
+              containerPort: {{ .Values.service.http.port }}
+            {{- end }}
           {{- end }}
+          env:
+            - name: "DATABASE_PATH"
+              value: /data/db
+            - name: "POWERDNS_API_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "pdns.secret.name" . }}
+                  key: {{ include "pdns.secret.key" . }}
           volumeMounts:
             - mountPath: /data
               name: data
-            - mountPath: /etc/pdns
-              name: config
             - mountPath: /zones
               name: zones
+          resources:
+            {{- toYaml .Values.pdns.resources | nindent 12 }}
       volumes:
         - name: data
           emptyDir: {}
-        - name: config
-          configMap:
-            name: {{ include "pdns.fullname" . }}-config
         - name: zones
           configMap:
-            name: {{ include "pdns.fullname" . }}-zones
+            name: {{ include "stateless-dns.fullname" . }}
```

This seems to lack `-zones` (or the CM has a leftover `-zones`)
```suggestion
            name: {{ include "stateless-dns.fullname" . }}-zones
```

---

Good catch!

It was the `configMap` the one which has a leftover.

Just for the sake of understanding better, As we are moving to a sidecar pattern, everything that has only one object (like one configMap, one secret, one clusterrol) uses the chart `fullname` (as of `include "stateless-dns.fullname" .`) instead of an unique one for PowerDNS lad the pdns deployment has.

Solved!